tests/tracer/test_propagation.py

```diff
@@ -3602,3 +3602,114 @@ def test_baggage_span_tags_wildcard():
+    headers = {}
+    with ddtracer.trace("parent") as parent:
+        with ddtracer.start_span("child", child_of=parent) as non_active_child:
+            assert non_active_child.context.sampling_priority is None  # No sampling yet
+            assert ddtracer.current_span() is not non_active_child  # Child is not active
+            with caplog.at_level(logging.DEBUG), pytest.warns() as warnings_list:
+                HTTPPropagator.inject(parent.context, headers, non_active_child)
+            # Sampling decision should be set on root span even when child is used for propagation
+            assert parent.context.sampling_priority is not None
+            assert non_active_child.context.sampling_priority is not None
+
+    assert any("sampled before propagating trace" in record.message for record in caplog.records), caplog.records
+    assert headers.get("x-datadog-sampling-priority") == str(
+        parent.context.sampling_priority
+    )  # Root span priority used
+    assert headers.get("x-datadog-parent-id") == str(
+        non_active_child.span_id
+    )  # Child span info propagated (non_active_span takes precedence)
+
+    # Should capture deprecation warning
+    assert len(warnings_list) == 1
+    assert "non_active_span parameter is deprecated" in str(warnings_list[0].message)
+
+
+def test_inject_context_and_span_different_trace_deprecated(caplog):
+    """Test injecting Context + non_active_span from completely different traces."""
+    headers = {}
+    with ddtracer.start_span("span1", child_of=None) as span1:
+        pass
+
+    with ddtracer.start_span("span2", child_of=None) as span2:
+        pass
+
+    with caplog.at_level(logging.DEBUG), pytest.warns() as warnings_list:
+        HTTPPropagator.inject(span1.context, headers, span2)
+
+    assert headers.get("x-datadog-parent-id") == str(span2.span_id)  # non_active_span takes precedence
+
+    # Should capture deprecation warning
+    assert len(warnings_list) == 1
+    assert "non_active_span parameter is deprecated" in str(warnings_list[0].message)
+
+
+def test_inject_context_without_sampling_priority_active_trace(caplog):
+    """Test injecting a Context without sampling priority when there's an active trace."""
+    headers = {}
+    with ddtracer.trace("test_span") as span:
+        context = span.context
+        assert context.sampling_priority is None  # No sampling decision yet
+        with caplog.at_level(logging.DEBUG):
+            HTTPPropagator.inject(context, headers)
+        assert span.context.sampling_priority is not None  # Sampling should be triggered automatically
+
+    assert any("sampled before propagating trace" in record.message for record in caplog.records), caplog.records
+    assert headers.get("x-datadog-sampling-priority") == str(
+        span.context.sampling_priority
+    )  # Headers include computed priority
+
+
+def test_inject_context_without_sampling_priority_inactive_trace(caplog):
```

Comment:
This behavior is problematic. It is likely breaking distributed tracing in a few integrations.